library.c

```diff
@@ -3315,10 +3339,12 @@ PHP_REDIS_API void redis_free_socket(RedisSock *redis_sock)
     if (redis_sock->host) {
         zend_string_release(redis_sock->host);
     }
-    if (redis_sock->subs) {
-        zend_hash_destroy(redis_sock->subs);
-        efree(redis_sock->subs);
-        redis_sock->subs = NULL;
+    for (i = 0; i < 3; ++i) {
```

Comment:
Given how you define subs, the sizeof(arr)/sizeof(arr[0]) trick will work.  Alternatively you could do something like:

```#define REDIS_SUBSCRIBE_BUCKETS 3```

The only reason is that it's not clear why it's `3` on this line of code, whereas it would be if using a define or calculating the size here.

```suggestion
    for (i = 0; i < sizeof(redis_sock->subs)/sizeof(redis_sock->subs[0]); ++i) {
```